presto-main/src/main/java/com/facebook/presto/execution/scheduler/SqlQueryScheduler.java

```diff
@@ -630,9 +633,17 @@ public void start()
 
     private List<SqlStageExecution> getStagesReadyForExecution()
     {
+        long runningPlanSections =
```

Comment:
The loop looks like "select a batch of stages ready for execution" -> "finish all selected stages" -> "select next batch". So, technically it always supposed to be 0.

Reply:
Do you think i should get rid of this then?  My concern is that it's not enforced anywhere that there are no running stages(and actually maybe makes sense to change now given that not all "ready" stages get scheduled at once).

Reply:
Actually I'm wrong. It can be more than 0. For example if some section has the scheduling finished, but hasn't finished the execution yet. Keep it.

Reply:
Another thought is to maintain the number of running sections as an AtomicLong. But maybe we don't need to do this.

Reply:
This is gonna be +1 state. The tree traversal should be fast. If we can avoid having another state - it should be easier to read.